src/rust/ide/view/graph-editor/src/lib.rs

```diff
@@ -2972,13 +3044,19 @@ fn new_graph_editor(app:&Application) -> GraphEditor {
     out.source.on_edge_only_source_not_set <+ out.on_edge_target_set_with_source_not_set._0();
     out.source.on_edge_only_source_not_set <+ out.on_edge_source_unset._0();
 
-    eval out.on_edge_source_set   (((id,_)) model.refresh_edge_color(*id));
-    eval out.on_edge_target_set   (((id,_)) model.refresh_edge_color(*id));
-    eval out.on_edge_source_unset (((id,_)) model.refresh_edge_color(*id));
-    eval out.on_edge_target_unset (((id,_)) model.refresh_edge_color(*id));
+    let neutral_color = styles_frp.get_color(theme::code::types::any::selection);
+    _eval <- all_with(&out.on_edge_source_set,&neutral_color,f!(((id,_),neutral_color)
+            model.refresh_edge_color(*id,neutral_color.into())));
+    _eval <- all_with(&out.on_edge_target_set,&neutral_color,f!(((id,_),neutral_color)
+            model.refresh_edge_color(*id,neutral_color.into())));
+    _eval <- all_with(&out.on_edge_source_unset,&neutral_color,f!(((id,_),neutral_color)
+            model.refresh_edge_color(*id,neutral_color.into())));
+    _eval <- all_with(&out.on_edge_target_unset,&neutral_color,f!(((id,_),neutral_color)
+            model.refresh_edge_color(*id,neutral_color.into())));
```

Comment:
There is a major issue with this code, but IMO easy to fix. Think what would happen when theme changes. The `neutral_color` will emit new event and thus all of these FRP nodes will fire (not only those, in other places as well), which means, that `model.refresh_edge_color` will be fired MANY times then. This is bad. After changing theme, we should refresh colors of edges only once per edge.